nixos/modules/services/web-apps/filesender.nix

```diff
@@ -0,0 +1,403 @@
+      };
+      port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = "Database port.";
+      };
+      name = mkOption {
+        type = types.str;
+        default = "filesender";
+        description = "Database name.";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "filesender";
+        description = "Database user.";
+      };
+      passwordFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/filesender-dbpassword";
+        description = ''
+          A file containing the password corresponding to
+          [](#opt-services.filesender.database.user).
+        '';
+      };
+    };
+    settings = mkOption {
+      type = lib.types.submodule {
+        freeformType = phpSettingsType;
+        options = {
+          site_url = mkOption {
+            type = types.str;
+            description = "Site URL. Used in emails, to build URLs for logging in, logging out, build URL for upload endpoint for web workers, to include scripts etc.";
+          };
+          admin = mkOption {
+            type = types.str;
+            description = "UIDs (as per the configured saml_uid_attribute) of FileSender administrators. Accounts with these UIDs can access the Admin page through the web UI. Separate multiple addresses with a comma (',').";
+          };
+          admin_email = mkOption {
+            type = types.str;
+            description = "Email address of FileSender administrator(s). Separate multiple addresses with a comma (','). Emails regarding disk full etc. are sent here. You should use a role-address here.";
+          };
+          storage_filesystem_path = mkOption {
+            type = types.nullOr types.str;
+            description = "When using storage type filesystem this is the absolute path to the file system where uploaded files are stored until they expire. Your FileSender storage root.";
+          };
+          log_facilities = mkOption {
+            type = phpSettingsType;
+            default = [
+              {
+                type = "syslog";
+              }
+            ];
+            description = "Defines where FileSender logging is sent. You can sent logging to a file, to syslog or to the default PHP log facility (as configured through your webserver's PHP module). The directive takes an array of one or more logging targets. Logging can be sent to multiple targets simultaneously. Each logging target is a list containing the name of the logging target and a number of attributes which vary per log target. See below for the exact definiation of each log target.";
+          };
+        };
+      };
+      default = { };
+      description = ''
+        Configuration options used by Filesender.
+        See [](https://docs.filesender.org/filesender/v2.0/admin/configuration/)
+        for available options.
+      '';
+    };
+
+    simpleSamlPhp = mkOption {
```

Comment:
Why are both `filesender` and `simpleSamlPhp` defined in the same file? Seems less readable to me.

Reply:
I think it's better because SimpleSAMLphp wants to be monolithic with target application, so I don't quite understand what other application may require it. PHP world is fun...

Reply:
@balsoft And I can't just plug new phpfpm for SimpleSAMLphp instance as I understand. So it's impossible to do separate module, right?

Reply:
Yeah, filesender wants shared phpfpm with simplesamlphp. So doing new module will just add more garbage, because it will not hold any logic.

Reply:
Why? Wouldn't it be possible to have a module which "adds" SimpleSAMLphp to (potentially multiple) `phpfpm`'s and `virtualHost`'s? This way it can be reused if some other PHP service requires it.

Reply:
I separated SimpleSAMLphp and Filesender modules.